src/app/static/src/tests/components/header/fileMenu.test.ts

```diff
@@ -331,11 +335,22 @@ describe("File menu", () => {
         expect(projectModal.props().openModal).toBe(false)
         projectModal.find(".btn").trigger("click");
         expect(mockLoadAction.mock.calls.length).toBe(1);
-        expect(mockGetProjects).toHaveBeenCalled()
         expect(projectModal.props().openModal).toBe(false)
     });
 
-    it("can open upload project modal as guest when file is uploaded", () => {
+    it("can get projects when user is logged in when file is uploaded", () => {
```

Comment:
Shouldn't this test, and the subsequent one, for guest user, be in `uploadNewProject.tests.ts`? that's where the code is that we're testing for! Even if we're rolling the tests for the two component together, won't you find that `getProjects` is called as soon as the parent component is mounted (without needing to upload file) - as we found the bug was happening as soon as the projects page was opened, without any user interaction

Reply:
`uploadNewProject.tests.ts` are tested already, I know it's an over kill but since we can't test this behaviour directly via a browser test, I felt it would be better to add tests in all cases where this behaviour is expected.

Reply:
OK - I think maybe I was looking at a slightly earlier commit, before you added the uploadNewProject test, so I'm happy that's in. And I'm still not sure this one tells us much, particularly since we don't need to open the modal to trigger (or not) the getProjects. But no harm to have this too!